packages/storybook/src/generators/configuration/util-functions.ts

```diff
@@ -533,3 +536,31 @@ export function rootFileIsTs(
     return tsConfiguration;
   }
 }
+
+export function getE2EProjectName(
+  tree: Tree,
+  mainProject: string
+): string | undefined {
+  let e2eProject: string;
+  forEachExecutorOptions(
+    tree,
+    '@nrwl/cypress:cypress',
+    (options, projectName) => {
+      if (e2eProject) {
+        return;
+      }
+      if (options['devServerTarget']) {
+        const { project, target } = parseTargetString(
+          options['devServerTarget']
+        );
+        if (
+          (project === mainProject && target === 'serve') ||
+          (project === mainProject && target === 'storybook')
+        ) {
+          e2eProject = projectName;
+        }
+      }
+    }
+  );
+  return e2eProject;
```

Comment:
@FrozenPandaz should we add some utils in @nrwl/devkit to handle this better? something like:

```
findProjectWithTarget(executor?: name, searchFn: (TargetConfiguration) => boolean)
```

Reply:
Yeah, we talked about this last week. The `forEachExecutorOptions` does this job at the moment. If we decide to add another function, I'll be happy to write it! :D

Reply:
My suggestion was `findBuildTarget`, or something similar, to get the name of various targets. But since the callback in `forEachExecutorOptions` essentially can be used for that, we said that it's better to keep the one.